client/app/scripts/superdesk-archive/module.js

```diff
@@ -134,14 +134,145 @@ define([
         };
     }
 
-    ArchiveService.$inject = ['desks', 'session'];
-    function ArchiveService(desks, session) {
+    ArchiveService.$inject = ['desks', 'session', 'api', '$q'];
+    function ArchiveService(desks, session, api, $q) {
+        /**
+         * Adds 'task' property to the article represented by item.
+         *
+         * @param {Object} item
+         */
         this.addTaskToArticle = function (item) {
             if ((!item.task || !item.task.desk) && desks.activeDeskId && desks.userDesks) {
                 var currentDesk = _.find(desks.userDesks._items, {_id: desks.activeDeskId});
                 item.task = {'desk': desks.activeDeskId, 'stage': currentDesk.incoming_stage, 'user': session.identity._id};
             }
         };
+
+        /**
+         * Returns the type of the item.
+         *
+         * @param {Object} item
+         */
+        this.getType = function(item) {
+            var itemType;
+
+            if (this.isLegal(item)) {
+                itemType = item._type;
+            } else if (item.state === 'spiked') {
+                itemType = 'spike';
+            } else if (item.state === 'ingested') {
+                itemType = 'ingest';
+            } else {
+                var isPublished = this.isPublished(item);
+
+                if (!isPublished || (isPublished && item.allow_post_publish_actions === true)) {
+                    itemType = 'archive';
+                } else if (isPublished && item.allow_post_publish_actions === false) {
+                    itemType = 'archived';
+                }
+            }
+
+            return itemType;
+        };
+
+        /**
+         * Returns true if the item is fetched from Legal Archive
+         *
+         * @param {Object} item
+         */
+        this.isLegal = function(item) {
+            return (angular.isDefined(item._type) && !_.isNull(item._type) && item._type === 'legal_archive');
+        };
+
+        /**
+         * Returns true if the state of the item is in one of the published states - Scheduled, Published, Corrected
+         * and Killed.
+         *
+         * @param {Object} item
+         */
+        this.isPublished = function(item) {
+            return _.contains(['published', 'killed', 'scheduled', 'corrected'], item.state);
+        };
+
+        /***
+         * Returns version history of the item.
+         *
+         * @param {Object} item
+         * @param {Object} desks deskService
+         * @param {String} historyType one of versions, operations
```

Comment:
I would just add the description of the method's return type to make it clear that it returns a promise (`'version history'` might be understood as some kind of an array of objects or  something).